multipartfile.go

```diff
@@ -22,94 +21,111 @@ const (
 	contentTypeHeader = "Content-Type"
 )
 
-var ErrPartOutsideParent = errors.New("file outside parent dir")
-var ErrPartInChildTree = errors.New("file in child tree")
+type multipartDirectory struct {
+	path   string
+	walker *multipartWalker
 
-// multipartFile implements Node, and is created from a `multipart.Part`.
-type multipartFile struct {
-	Node
+	// part is the part describing the directory. It's nil when implicit.
+	part *multipart.Part
+}
 
-	part      *multipart.Part
-	reader    *peekReader
-	mediatype string
+type multipartWalker struct {
```

Comment:
Replaced `peakReader` with this so we don't have to keep putting the parts back.